iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/SlackBusImpl.java

```diff
@@ -0,0 +1,102 @@
+package com.powsybl.iidm.network.extensions;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.VoltageLevel;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public class SlackBusImpl extends AbstractExtension<VoltageLevel> implements SlackBus {
+
+    private final NodeBreakerViewImpl nodeBreakerView;
+    private final BusBreakerViewImpl busBreakerView;
+    private final BusViewImpl busView;
+
+    private static final class NodeBreakerViewImpl implements SlackBus.NodeBreakerView {
+        private final int node;
+
+        public NodeBreakerViewImpl(int node) {
+            this.node = node;
+        }
+
+        @Override
+        public int getNode() {
+            return this.node;
+        }
+    }
+
+    private static final class BusBreakerViewImpl implements SlackBus.BusBreakerView {
+        private final Bus bus;
+
+        public BusBreakerViewImpl(Bus busBusBreakerView) {
+            this.bus = busBusBreakerView;
+        }
+
+        @Override
+        public Bus getBus() {
+            return this.bus;
+        }
+    }
+
+    private static final class BusViewImpl implements SlackBus.BusView {
+        private final Bus bus;
+
+        private BusViewImpl(Bus busBusView) {
+            this.bus = busBusView;
+        }
+
+        @Override
+        public Bus getBus() {
+            return this.bus;
+        }
+    }
+
+    SlackBusImpl(String busId, VoltageLevel vl) {
+        super(vl);
+        if (busId == null) {
+            throw new PowsyblException("busId needs to be set for a SlackBus in VoltageLevel.BusBreakerView");
+        }
+        Bus busBusBreakerView = vl.getBusBreakerView().getBus(busId);
+        Bus busBusView = vl.getBusView().getMergedBus(busId);
+        this.nodeBreakerView = null;
+        this.busBreakerView = new BusBreakerViewImpl(busBusBreakerView);
+        this.busView = new BusViewImpl(busBusView);
+    }
+
+    SlackBusImpl(int node, VoltageLevel vl) {
+        super(vl);
+        if (node == SlackBusAdderImpl.NODE_UNINITIALIZED) {
+            throw new PowsyblException("node needs to be set for a SlackBus in VoltageLevel.NodeBreakerView");
+        }
+        Terminal terminal = vl.getNodeBreakerView().getOptionalTerminal(node).get();
+        Bus busBusBreakerView = terminal.getBusBreakerView().getBus();
+        Bus busBusView = terminal.getBusView().getBus();
+        this.busBreakerView = new BusBreakerViewImpl(busBusBreakerView);
+        this.nodeBreakerView = new NodeBreakerViewImpl(node);
+        this.busView = new BusViewImpl(busBusView);
```

Comment:
This is a really bad idea to store the bus in the view: in a Node/Breaker Voltage level, these buses are calculated automatically, so new buses will be created when the topology changes. For Bus/Breaker Voltage Level, the bus of the bus/breaker view is configured, so it could be OK to keep it, but it's not for the bus view.